src/client-functions/selector-builder/index.js

```diff
@@ -144,6 +146,22 @@ export default class SelectorBuilder extends ClientFunctionBuilder {
         if (snapshot && !this.options.counterMode) {
             this._addBoundArgsSelectorGetter(snapshot, selectorArgs);
             createSnapshotMethods(snapshot);
+
+            if (this.options.customMethods) {
```

Comment:
Why this code duplication? Also, do we really need to add methods to snapshot? I guess it should be just Selector methods